test/zero-downtime/API_test.go

```diff
@@ -0,0 +1,209 @@
+		Expect(suite.T()).Status(http.StatusOK).Assert(jsonpath.Contains(`$.scopes`, "keptn-default")).End()
+
+}
+
+func (suite *TestSuiteAPI) Test_Configuration_Service() {
+
+	started := time.Now()
+	apiURL := suite.keptnAPIURL + "/configuration-service/v1"
+
+	api := apitest.New("Test configuration-service: not existing project").
+		Observe(func(res *http.Response, req *http.Request, apiTest *apitest.APITest) {
+			suite.logResult(res, apiTest, http.StatusNotFound, started)
+		}).EnableNetworking(getClient(1))
+
+	req := api.Get(apiURL + "/project/unexisting-project/resource").
+		Headers(map[string]string{"x-token": suite.token}).
+		Expect(suite.T()).
+		Status(http.StatusNotFound)
+
+	if res, err := testutils.CompareServiceNameWithDeploymentName("configuration-service", "configuration-service"); err == nil && res {
+		req.Assert(jsonpath.Equal(`$.message`, "Project does not exist")).End()
+	} else {
+		req.Assert(jsonpath.Equal(`$.message`, "Could not find credentials for upstream repository")).End()
+	}
+
+}
+
+func (suite *TestSuiteAPI) Test_ControlPlane() {
+	started := time.Now()
+	apiURL := suite.keptnAPIURL + "/controlPlane/v1"
+
+	api := apitest.New("Test control-plane: check uniform").EnableNetworking(getClient(1)).
+		Observe(func(res *http.Response, req *http.Request, apiTest *apitest.APITest) {
+			suite.logResult(res, apiTest, http.StatusOK, started)
+		})
+
+	api.Get(apiURL+"/uniform/registration").Query("name", "lighthouse-service").
+		Headers(map[string]string{"x-token": suite.token}).
+		Expect(suite.T()).Status(http.StatusOK).Assert(jsonpath.Equal(`$[0].name`, "lighthouse-service")).End()
+
+}
+
+func (suite *TestSuiteAPI) Test_MongoDB() {
+	started := time.Now()
+	apiURL := suite.keptnAPIURL + "/mongodb-datastore"
+
+	api := apitest.New("Test mongo-datastore: not existing project").EnableNetworking(getClient(1)).
+		Observe(func(res *http.Response, req *http.Request, apiTest *apitest.APITest) {
+			suite.logResult(res, apiTest, http.StatusOK, started)
+		})
+
+	api.Get(apiURL+"/event").Query("project", "keptn").Query("pageSize", "20").
+		Headers(map[string]string{"x-token": suite.token}).
+		Expect(suite.T()).Status(http.StatusOK).Body(`{"events":[], "pageSize":20}`).End()
+
+}
+
+func (suite *TestSuiteAPI) logResult(res *http.Response, apiTest *apitest.APITest, expected int, started time.Time) {
+	atomic.AddUint64(&(suite.env.TotalAPICalls), 1)
+	finished := time.Now()
+	//remove headers cookies and auth from request
+	censoredReq := strings.Split(strings.Split(fmt.Sprintf("%+v", *apiTest.Request()), "{interceptor:<nil>")[1], "headers:")[0]
+
+	suite.T().Log("\n", "Test API configuration: ", censoredReq)
+	suite.T().Logf("Response Body: %v", res.Body)
+	suite.T().Logf("Duration: %s", finished.Sub(started))
+	if res.StatusCode != expected {
+		atomic.AddUint64(&suite.env.FailedAPICalls, 1)
+	} else {
+		atomic.AddUint64(&suite.env.PassedAPICalls, 1)
+	}
+}
+
+func getClient(sec time.Duration) *http.Client {
```

Comment:
maybe also moving this to `test/zero-downtime/testsuite_zerodowntime_test.go` ? To make it reusable also in other ZD tests in the future if needed

Reply:
not sure if it is needed this is specifically for HTTP probes and I would leave them in the API suite

Reply:
ok